src/services/completions.ts

```diff
@@ -2616,7 +2616,8 @@ namespace ts.Completions {
 
             // Need to insert 'this.' before properties of `this` type, so only do that if `includeInsertTextCompletions`
             if (preferences.includeCompletionsWithInsertText && scopeNode.kind !== SyntaxKind.SourceFile) {
-                const thisType = typeChecker.tryGetThisTypeAt(scopeNode, /*includeGlobalThis*/ false);
+                const thisType = typeChecker.tryGetThisTypeAt(scopeNode, /*includeGlobalThis*/ false,
+                    isClassLike(scopeNode.parent) ? scopeNode : getThisContainer(scopeNode, /*includeArrowFunctions*/ false));

```

Comment:
Also, would it make more sense to move this logic into `getThisContainer`? Is this something that's just broken everywhere?

Reply:
@jakebailey I think this case is more related to the `completions`. `scopeNode` refers to `PropertyDeclaration` and the parent for it is  `ClassDeclaration` - `getThisContainer` doesn't treat class-like declarations like `this` container - which seems correct. So for this case `container` will be `SourceFile` and `tryGetThisTypeAt` returns `undefined`. So for the case where this is located in `PropertyDeclaration` and the parent is class-like we can use `PropertyDeclaration` as a container to get a type of `this`.

```ts
if (isClassLike(container.parent)) {
    const symbol = getSymbolOfNode(container.parent);
    const type = isStatic(container) ? getTypeOfSymbol(symbol) : (getDeclaredTypeOfSymbol(symbol) as InterfaceType).thisType!;
    return getFlowTypeOfReference(node, type);
}
```
 Does that make sense?

Reply:
That sounds reasonable, yeah.